File: DWTTest.java
import java.io.PrintStream;
import java.util.Random;

import test.testDWTSimulationGenerator;

import com.google.common.base.Stopwatch;

public class DWTTest {
	public static void main(String[] args) throws Exception {
		final testDWTSimulationGenerator gen = new testDWTSimulationGenerator(true, true);
		try (final PrintStream inputFile = new PrintStream("TestDWTSimInputData.txt"); final PrintStream outputFile = new PrintStream("TestDWTSimOuputData.txt");) {
			final Random r = new Random();
			gen.initConstants();
			final int width = 1024;
			final int max = width * width;
			final long[] data = new long[max];
			for (int i = 0; i < data.length; i++) {
				data[i] = r.nextInt() & 0x3ff;
				inputFile.printf("%03x%n", data[i]);
			}
			// System.out.println(Arrays.toString(data));
			while (true) {
				final Stopwatch sw = new Stopwatch();
				gen.de_henningholm_da_DWTSimulation_width_tile = width;
				gen.de_henningholm_da_DWTSimulation_height_tile = width;
				gen.de_henningholm_da_DWTSimulation_count_resolutions = 1;
				// System.arraycopy(data, 0,
				// gen.de_henningholm_da_DWTSimulation_VFBC_SimulationInst_memory,
				// 0, max);
				gen.de_henningholm_da_DWTSimulation_address_destination = max * 4;
				gen.de_henningholm_da_DWTSimulation_clk = 1;
				gen.de_henningholm_da_DWTSimulation_rst = 1;
				gen.run();
				gen.de_henningholm_da_DWTSimulation_rst = 0;
				sw.start();
				gen.de_henningholm_da_DWTSimulation_dwt_start = 1;
				gen.run();
				do {
					gen.run();
				} while (gen.de_henningholm_da_DWTSimulation_dwt_done == 0);
				System.out.println("Total:" + sw);
				// for (final long element :
				// gen.de_henningholm_da_DWTSimulation_VFBC_SimulationInst_memory)
				// {
				// outputFile.printf("%03x%n", element);
				// }
			}
		}
	}
}

File: dwtTest.c
//
//  main.c
//  DWTTest
//
//  Created by Karsten Becker on 05.08.14.
//  Copyright (c) 2014 PSHDL. All rights reserved.
//

#include <stdio.h>
#include <stdbool.h>
#include <stdint.h>
#include <time.h>
#include "pshdl_de_henningholm_da_DWTSimulation_sim.h"


int main(int argc, const char * argv[])
{
    int width=1024;
    int max=width*width;
    // insert code here...
    printf("Hello, World!\n");
    pshdl_sim_setDisableEdges(true);
    pshdl_sim_setDisableRegOutputlogic(true);

    pshdl_sim_initConstants();
    printf("Index of clk: %d\n", pshdl_sim_getIndex("de.henningholm.da.DWTSimulation.clk"));
    de_henningholm_da_DWTSimulation_width_tile = width;
    de_henningholm_da_DWTSimulation_height_tile = width;
    de_henningholm_da_DWTSimulation_count_resolutions = 1;
    de_henningholm_da_DWTSimulation_address_destination = max * 4;
    int count=0;
    while(true){
        if (count++>5)
            return 0;
        clock_t start = clock() / (CLOCKS_PER_SEC / 1000);
        de_henningholm_da_DWTSimulation_clk = 1;
        de_henningholm_da_DWTSimulation_rst = 1;
        pshdl_sim_run();
        de_henningholm_da_DWTSimulation_rst = 0;
        de_henningholm_da_DWTSimulation_dwt_start = 1;
        pshdl_sim_run();
        do {
            pshdl_sim_run();
        } while (de_henningholm_da_DWTSimulation_dwt_done == 0);
        clock_t end = clock() / (CLOCKS_PER_SEC / 1000);
        printf("Took %lu\n", end-start);
    }
    return 0;
}

